Report Tab causes UI crash on startup on latest MacOS.

GAPID Version: 1.4.0:5f4620616cbea20dca03ecf0ea29dfcd824114e3
OS: mac os x x86_64

To reproduce: Start GAPID on MacOSX 10.14.3 

java.lang.NullPointerException
	at com.google.gapid.widgets.TabArea.layoutComputeSize(TabArea.java:344)
	at com.google.gapid.widgets.TabArea$1.computeSize(TabArea.java:68)
	at org.eclipse.swt.widgets.Composite.computeSize(Composite.java:229)
	at org.eclipse.swt.layout.FormData.computeSize(FormData.java:120)
	at org.eclipse.swt.layout.FormData.getWidth(FormData.java:149)
	at org.eclipse.swt.layout.FormData.getRightAttachment(FormData.java:250)
	at org.eclipse.swt.layout.FormLayout.computeWidth(FormLayout.java:271)
	at org.eclipse.swt.layout.FormLayout.layout(FormLayout.java:338)
	at org.eclipse.swt.layout.FormLayout.computeSize(FormLayout.java:245)
	at org.eclipse.swt.widgets.Composite.computeSize(Composite.java:229)
	at org.eclipse.swt.layout.GridData.computeSize(GridData.java:494)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:224)
	at org.eclipse.swt.layout.GridLayout.computeSize(GridLayout.java:167)
	at org.eclipse.swt.widgets.Composite.computeSize(Composite.java:229)
	at org.eclipse.swt.custom.StackLayout.computeSize(StackLayout.java:101)
	at org.eclipse.swt.widgets.Composite.computeSize(Composite.java:229)
	at org.eclipse.swt.custom.StackLayout.computeSize(StackLayout.java:101)
	at org.eclipse.swt.widgets.Composite.computeSize(Composite.java:229)
	at org.eclipse.swt.layout.GridData.computeSize(GridData.java:494)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:224)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:200)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1267)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1273)
	at org.eclipse.swt.widgets.Composite.setLayoutDeferred(Composite.java:1143)
	at org.eclipse.swt.widgets.Display.runDeferredLayouts(Display.java:4140)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3719)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:821)
	at org.eclipse.jface.window.Window.open(Window.java:797)
	at com.google.gapid.views.ErrorDialog.showErrorDialog(ErrorDialog.java:161)
	at com.google.gapid.views.ErrorDialog.showErrorDialog(ErrorDialog.java:62)
	at com.google.gapid.Main$UI.lambda$registerWindowExceptionHandler$0(Main.java:113)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:825)
	at org.eclipse.jface.window.Window.open(Window.java:797)
	at com.google.gapid.Main$UI.show(Main.java:119)
	at com.google.gapid.Main.main(Main.java:68)

org.eclipse.swt.SWTException: Failed to execute runnable (java.lang.NullPointerException)
	at org.eclipse.swt.SWT.error(SWT.java:4595)
	at org.eclipse.swt.SWT.error(SWT.java:4510)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:188)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4068)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3735)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:821)
	at org.eclipse.jface.window.Window.open(Window.java:797)
	at com.google.gapid.Main$UI.show(Main.java:119)
	at com.google.gapid.Main.main(Main.java:68)
Caused by: java.lang.NullPointerException
	at org.eclipse.swt.widgets.Control.calculateVisibleRegion(Control.java:784)
	at org.eclipse.swt.widgets.Control.getVisibleRegion(Control.java:2067)
	at org.eclipse.swt.widgets.Control.isObscured(Control.java:2336)
	at org.eclipse.swt.widgets.Scrollable.scrollClipViewToPoint(Scrollable.java:351)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:6461)
	at org.eclipse.swt.internal.cocoa.OS.objc_msgSend(Native Method)
	at org.eclipse.swt.internal.cocoa.NSScrollView.setDocumentView(NSScrollView.java:77)
	at org.eclipse.swt.widgets.Scrollable.setZOrder(Scrollable.java:400)
	at org.eclipse.swt.widgets.Control.createWidget(Control.java:1035)
	at org.eclipse.swt.widgets.Scrollable.createWidget(Scrollable.java:173)
	at org.eclipse.swt.widgets.Text.createWidget(Text.java:591)
	at org.eclipse.swt.widgets.Control.<init>(Control.java:123)
	at org.eclipse.swt.widgets.Scrollable.<init>(Scrollable.java:76)
	at org.eclipse.swt.widgets.Text.<init>(Text.java:136)
	at com.google.gapid.views.ReportView.<init>(ReportView.java:85)
	at com.google.gapid.GraphicsTraceView$MainTab.lambda$getFolders$1(GraphicsTraceView.java:224)
	at com.google.gapid.widgets.TabArea$Folder.addTab(TabArea.java:525)
	at com.google.gapid.widgets.TabArea$Folder.<init>(TabArea.java:509)
	at com.google.gapid.widgets.TabArea.<init>(TabArea.java:80)
	at com.google.gapid.GraphicsTraceView$1.createBottomControl(GraphicsTraceView.java:89)
	at com.google.gapid.widgets.FixedTopSplitter.<init>(FixedTopSplitter.java:39)
	at com.google.gapid.GraphicsTraceView$1.<init>(GraphicsTraceView.java:81)
	at com.google.gapid.GraphicsTraceView.<init>(GraphicsTraceView.java:81)
	at com.google.gapid.MainWindow.lambda$initMainUi$0(MainWindow.java:102)
	at com.google.gapid.widgets.LoadablePanel.<init>(LoadablePanel.java:56)
	at com.google.gapid.MainWindow.initMainUi(MainWindow.java:101)
	at com.google.gapid.Main$UI.uiStartup(Main.java:154)
	at com.google.gapid.Main$UI.lambda$scheduleIfStillOpen$5(Main.java:181)
	at com.google.gapid.widgets.Widgets.ifNotDisposed(Widgets.java:119)
	at com.google.gapid.widgets.Widgets.lambda$scheduleIfNotDisposed$1(Widgets.java:147)
	at com.google.gapid.widgets.Widgets.lambda$schedule$0(Widgets.java:139)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	... 6 more

--- a/gapic/src/main/com/google/gapid/views/ReportView.java
+++ b/gapic/src/main/com/google/gapid/views/ReportView.java
@@ -66,7 +66,8 @@ public class ReportView extends Composite implements Tab, Capture.Listener, Repo
   private final MessageProvider messages = new MessageProvider();
   private final LoadablePanel<SashForm> loading;
   private final TreeViewer viewer;
-  private final Text reportDetails;
+  private final Composite detailsGroup;
+  private Text reportDetails;
 
   public ReportView(Composite parent, Models models, Widgets widgets) {
     super(parent, SWT.NONE);
@@ -81,8 +82,7 @@ public class ReportView extends Composite implements Tab, Capture.Listener, Repo
     ViewLabelProvider labelProvider = new ViewLabelProvider(viewer, widgets.theme, messages);
     viewer.setLabelProvider(labelProvider);
 
-    Composite detailsGroup = Widgets.createGroup(splitter, "Details");
-    reportDetails = new Text(detailsGroup, SWT.MULTI | SWT.READ_ONLY | SWT.H_SCROLL | SWT.V_SCROLL);
+    detailsGroup = Widgets.createGroup(splitter, "Details");
 
     splitter.setWeights(models.settings.reportSplitterWeights);
 
@@ -110,12 +110,20 @@ public class ReportView extends Composite implements Tab, Capture.Listener, Repo
           item = item.getParentItem();
         }
         if (item != null) {
-          reportDetails.setText(((Group)item.getData()).name);
-          reportDetails.requestLayout();
+          getDetails().setText(((Group)item.getData()).name);
+          getDetails().requestLayout();
         }
       }
     });
     addListener(SWT.Dispose, e -> models.settings.reportSplitterWeights = splitter.getWeights());
+  }
+
+  private Text getDetails() {
+    // Lazy init'ed due to https://github.com/google/gapid/issues/2624
+    if (reportDetails == null) {
+      reportDetails = new Text(detailsGroup, SWT.MULTI | SWT.READ_ONLY | SWT.H_SCROLL | SWT.V_SCROLL);
+    }
+    return reportDetails;
   }
 
   @Override
@@ -149,7 +157,7 @@ public class ReportView extends Composite implements Tab, Capture.Listener, Repo
   @Override
   public void onReportLoaded() {
     messages.clear();
-    reportDetails.setText("");
+    getDetails().setText("");
     if (models.reports.isLoaded()) {
       updateReport();
     } else {

